cmd/jujud/agent/bootstrap.go

```diff
@@ -374,19 +452,25 @@ func (c *BootstrapCommand) startMongo(addrs []network.Address, agentConfig agent
 		net.JoinHostPort("localhost", fmt.Sprint(servingInfo.StatePort)),
 	}
 
-	logger.Debugf("calling ensureMongoServer")
-	ensureServerParams, err := cmdutil.NewEnsureServerParams(agentConfig)
-	if err != nil {
-		return err
-	}
-	_, err = cmdutil.EnsureMongoServer(ensureServerParams)
-	if err != nil {
-		return err
+	if !isCAAS {
+		// TODO(bootstrap): recheck if the cert, keys, etc have been updated after k8s template generated.
+		logger.Debugf("calling ensureMongoServer")
+		ensureServerParams, err := cmdutil.NewEnsureServerParams(agentConfig)
+		if err != nil {
+			return err
+		}
+		_, err = cmdutil.EnsureMongoServer(ensureServerParams)
+		if err != nil {
+			return err
+		}
 	}
 
-	peerAddr := mongo.SelectPeerAddress(addrs)
-	if peerAddr == "" {
-		return fmt.Errorf("no appropriate peer address found in %q", addrs)
+	peerAddr := "127.0.0.1"
```

Comment:
Add a todo to support HA

Reply:
ah, just removed here, coz it's already ensured `getAddressesForMongo`.
And it's not related with HA. 
Because `bootstrap-state` `only` runs on machine-0.